File: app/src/main/java/com/asmat/rolando/popularmovies/MainActivity.java
package com.asmat.rolando.popularmovies;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;

public class MainActivity extends AppCompatActivity {

    private RecyclerView mMoviesGrid;
    private MoviesGridAdapter mMoviesGridAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // Get reference to layout elements
        mMoviesGrid = (RecyclerView) findViewById(R.id.movie_grid_rv);
        // Grid will not be of fixed size
        mMoviesGrid.setHasFixedSize(false);
        // Set layout manager: grid layout manager with 2 columns
        RecyclerView.LayoutManager layoutManager = new GridLayoutManager(this, 2);
        mMoviesGrid.setLayoutManager(layoutManager);
        // Set adapter
        mMoviesGridAdapter = new MoviesGridAdapter();
        mMoviesGrid.setAdapter(mMoviesGridAdapter);
    }

}

File: app/src/main/java/com/asmat/rolando/nightwing/main/MainActivity.kt
package com.asmat.rolando.nightwing.main

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.asmat.rolando.nightwing.R
import com.google.android.material.bottomnavigation.BottomNavigationView

/**
 * Main screen the user lands on upon launch.
 */
class MainActivity : AppCompatActivity() {

    private lateinit var navController: NavController
    private lateinit var appBarConfiguration: AppBarConfiguration

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val navHostFragment = supportFragmentManager.findFragmentById(
            R.id.nav_host_fragment
        ) as NavHostFragment
        navController = navHostFragment.navController

        // Setup the bottom navigation view with navController
        val bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottomNavigationView)
        bottomNavigationView.setupWithNavController(navController)

        // Setup the ActionBar with navController and 3 top level destinations
        appBarConfiguration = AppBarConfiguration(
            setOf(R.id.homeScreen, R.id.searchScreen,  R.id.savedTab)
        )
        navController.addOnDestinationChangedListener { _, destination, _ ->
            if (destination.label == "Movie Details" || destination.label == "Cast Details" || destination.label == "TV Show Details") {
                supportActionBar?.hide()
            } else {
                supportActionBar?.show()
            }
        }
        setupActionBarWithNavController(navController, appBarConfiguration)
    }

    override fun onSupportNavigateUp(): Boolean {
        return navController.navigateUp(appBarConfiguration)
    }
}